base/commands/migration/migration_stages.go

```diff
@@ -233,6 +241,46 @@ func fetchMigrationStatus(ctx context.Context, ci *hazelcast.ClientInternal, mig
 	return strings.TrimSuffix(strings.TrimPrefix(string(r.(serialization.JSON)), `"`), `"`), nil
 }
 
+func fetchOverallProgress(ctx context.Context, ci *hazelcast.ClientInternal, migrationID string) (time.Duration, float32, error) {
+	q := fmt.Sprintf(`SELECT JSON_QUERY(this, '$.remainingTime'), JSON_QUERY(this, '$.completionPercentage') FROM %s WHERE __key='%s'`, StatusMapName, migrationID)
+	res, err := ci.Client().SQL().Execute(ctx, q)
+	if err != nil {
+		return 0, 0, err
+	}
+	it, err := res.Iterator()
+	if err != nil {
+		return 0, 0, err
+	}
```

Comment:
This can be refactored to a function.

Reply:
I think it is should be ok like that. It is not reused.